src/main/java/com/ldtteam/structures/helpers/BlueprintIterator.java

```diff
@@ -0,0 +1,178 @@
+        if (this.progressPos.getX() >= this.size.getX())
+        {
+            this.progressPos.setPos(0, this.progressPos.getY(), this.progressPos.getZ() + 1);
+            if (this.progressPos.getZ() >= this.size.getZ())
+            {
+                this.progressPos.setPos(this.progressPos.getX(), this.progressPos.getY() + 1, 0);
+                if (this.progressPos.getY() >= this.size.getY())
+                {
+                    this.reset();
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Decrement progressPos.
+     *
+     * @return false if progressPos can't be decremented any more.
+     */
+    public boolean decrement()
+    {
+        if (this.progressPos.equals(NULL_POS))
+        {
+            this.progressPos.setPos(this.size.getX(), this.size.getY() - 1, this.size.getZ() - 1);
+        }
+
+        this.progressPos.setPos(this.progressPos.getX() - 1, this.progressPos.getY(), this.progressPos.getZ());
+        if (this.progressPos.getX() <= -1)
+        {
+            this.progressPos.setPos(this.size.getX() - 1, this.progressPos.getY(), this.progressPos.getZ() - 1);
+            if (this.progressPos.getZ() <= -1)
+            {
+                this.progressPos.setPos(this.progressPos.getX(), this.progressPos.getY() - 1, this.size.getZ() - 1);
+                if (this.progressPos.getY() <= -1)
+                {
+                    this.reset();
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    public boolean increment(final Predicate<BlueprintPositionInfo> blockInfoPredicate, final boolean includeEntities)
+    {
+        int count = 0;
+        do
+        {
+            final BlockPos worldPos = structureHandler.getPosition().subtract(structureHandler.getBluePrint().getPrimaryBlockOffset()).add(progressPos);
+            final BlueprintPositionInfo info = structureHandler.getBluePrint().getBluePrintPositionInfo(progressPos, includeEntities);
+            if(!increment())
+            {
+                return false;
+            }
+            else if (BlockUtils.areBlockStatesEqual(structureHandler.getBluePrint().getBlockInfoAsMap().get(progressPos).getState(), structureHandler.getWorld().getBlockState(worldPos)))
+            {
+                structureHandler.triggerSuccess(progressPos);
+            }
+            else if (!blockInfoPredicate.test(info))
+            {
+                return true;
+            }
+        }
+        while (count < Structurize.getConfig().getCommon().maxBlocksChecked.get());
+
+        return true;
+    }
+
+
+    public boolean decrement(final Predicate<BlueprintPositionInfo> blockInfoPredicate, final boolean includeEntities)
+    {
+        int count = 0;
+        do
+        {
+            final BlockPos worldPos = structureHandler.getPosition().subtract(structureHandler.getBluePrint().getPrimaryBlockOffset()).add(progressPos);
```

Comment:
isnt structureHandler.getPosition() our world pos of the primary offset(anchor) already?

Reply:
It is. So worldPos.subtract(primaroffset) = 0,0,0 pos of schematic + add progress = pos

Reply:
ah makes sense